tests/integrations/raftstore/test_replica_read.rs

```diff
@@ -507,3 +509,64 @@ fn test_read_local_after_snapshpot_replace_peer() {
     let exp_value = resp.get_responses()[0].get_get().get_value();
     assert_eq!(exp_value, b"v3");
 }
+
+/// The case checks if a malformed request should not corrupt the leader's read queue.
+#[test]
+fn test_malformed_read_index() {
+    let mut cluster = new_node_cluster(0, 3);
+    configure_for_lease_read(&mut cluster, Some(50), None);
+    cluster.cfg.raft_store.raft_log_gc_threshold = 12;
+    cluster.cfg.raft_store.raft_log_gc_count_limit = 12;
+    cluster.cfg.raft_store.hibernate_regions = true;
+    cluster.cfg.raft_store.check_leader_lease_interval = ReadableDuration::hours(10);
+    let pd_client = Arc::clone(&cluster.pd_client);
+    pd_client.disable_default_operator();
+
+    let region_id = cluster.run_conf_change();
+    pd_client.must_add_peer(region_id, new_peer(2, 2));
+    pd_client.must_add_peer(region_id, new_peer(3, 3));
```

Comment:
How about just `cluster.run()`

Reply:
Using `run_conf_change` is on purpose to speed up initialization. The lease is increased, so the election timeout will be long when there are multiple peers.